contracts/proxy/beacon/BeaconProxy.sol

```diff
@@ -12,8 +12,15 @@ import {ERC1967Utils} from "../ERC1967/ERC1967Utils.sol";
  *
  * The beacon address is stored in storage slot `uint256(keccak256('eip1967.proxy.beacon')) - 1`, so that it doesn't
  * conflict with the storage layout of the implementation behind the proxy.
+ *
+ * CAUTION: The beacon address can only be set once during construction, and cannot be changed afterwards.
+ * You must ensure that you either control the beacon, or trust the beacon to not upgrade the implementation maliciously.
  */
 contract BeaconProxy is Proxy {
+    // An immutable address for the beacon to avoid unnecessary SLOADs before each delegate call.
+    // This must be private to avoid generating a getter function that could clash with a function from the implementation.
```

Comment:
it could be internal. should it ?

Reply:
If we are keeping `getBeacon` as an internal function, the variable should be private so it is accessed through the getter.

Reply:
Which getter ?

Reply:
`getBeacon`

Reply:
I was confused about your other comment in this PR, I thought getBeacon was a getter in the BeaconProxy.

Reply:
Its two different thing:
- In the `ERC1967Utils` library there is a getter to read the storage slot, that I think we should keep is someone ever needs to read that (for example to re-implement a storage based `BeaconProxy`)
- In the `BeaconProxy` (with immutable storage) I think it would be nice to have a possible access to the immutable value, which can be done in a safe way by making the immutable variable internal.

Reply:
If we defined an internal function instead of making the variable internal it would be possible to do the storage variant by overriding, is that something we're interested in?

Reply:
Why not!

The parent's constructor will still write the value to immutable storage, but I guess its ok. Maybe if the immutable storage is not used anywhere that is optimized !

Reply:
Resolving as there is already an internal function `_getBeacon` that is being used, this means we don't need to make the variable internal.